src/app.js

```diff
@@ -277,6 +721,117 @@ fluid.onUncaughtException.addListener(function (err) {
 }, "gpii.app", "last");
 
 /**
+ * Resizes the PCP window and positions it appropriately based on the new height
+ * of its content. Makes sure that the window is no higher than the available
+ * height of the work area in the primary display. The window will not be resized
+ * if its current height is the same as the new height. This behaviour can be
+ * overridden using the `forceResize` parameter.
+ * @param pcp {Object} A `gpii.app.pcp` instance.
+ * @param contentHeight {Number} The new height of the BrowserWindow's content.
+ * @param minHeight {Number} The minimum height which the BrowserWindow must have.
+ * @param forceResize {Boolean} Whether to resize the window even if the current
+ * height of the `BrowserWindow` is the same as the new one. Useful when screen
+ * DPI is changed as a result of the application of a user's preferences.
+ */
+gpii.app.pcp.resize = function (pcp, contentHeight, minHeight, forceResize) {
+    var pcpWindow = pcp.pcpWindow,
+        wasShown = pcp.isShown(),
+        screenSize = electron.screen.getPrimaryDisplay().workAreaSize,
+        windowSize = pcpWindow.getSize(),
+        windowWidth = windowSize[0],
+        initialHeight = windowSize[1],
+        windowHeight = Math.min(screenSize.height, Math.max(contentHeight, minHeight));
+
+    if (initialHeight === windowHeight && !forceResize) {
+        return;
+    }
+
+    pcpWindow.setSize(windowWidth, windowHeight);
+
+    if (wasShown) {
+        pcp.show();
+    }
+};
+
+/**
+ * Handles logic for the PCP window.
+ */
+fluid.defaults("gpii.app.pcp", {
+    gradeNames: "fluid.modelComponent",
+
+    model:  {
+        keyedInUserToken: null
+    },
+
+    /*
+     * Options to be passed to the Electron `BrowserWindow`.
+     * Only not reserved options can be altered. See `reservedAttrs`.
+     */
+    attrs: {
+        width: 500,
+        height: 600,
+        show: true,
+        frame: false,
+        fullscreenable: false,
+        resizable: false,
+        alwaysOnTop: true,
+        skipTaskbar: true,
+        backgroundColor: "transparent"
+    },
+
+    members: {
+        pcpWindow: {
```

Comment:
This is simple enough to be usefully expressed in the compact format - 
pcpWindow: "@expand:gpii.app.pcp.makePCPWindow({that}.options.attrs)"